### PR TITLE
fix: migrated savings,update,user and loan fragments to viewbinding

### PR DESCRIPTION
**Pull request by PratyushSingh07**
fixes #2142 

https://github.com/openMF/mifos-mobile/assets/90026952/924fd589-51b9-4680-b2aa-d97319ddea8f

Please Add Screenshots If there are any UI changes.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by PratyushSingh07** [comment]
Hey @gururani-abhishek do review this

File: app/src/main/java/org/mifos/mobile/ui/fragments/SavingsAccountApplicationFragment.kt [hunk 1]
```diff
@@ -145,12 +131,11 @@ class SavingsAccountApplicationFragment : BaseFragment(), SavingsAccountApplicat
     private fun updateSavingAccount() {
         val payload = SavingsAccountUpdatePayload()
         payload.clientId = template?.clientId?.toLong()
-        payload.productId = productIdList.indexOf(productIdField?.text.toString())
+        payload.productId = productIdList.indexOf(binding.productIdField?.text.toString())
```
Comment (gururani-abhishek) [inline app/src/main/java/org/mifos/mobile/ui/fragments/SavingsAccountApplicationFragment.kt]: In "binding.productIdField?.text" please remove the unnecessary safe-call.

File: app/src/main/java/org/mifos/mobile/ui/fragments/SavingsMakeTransferFragment.kt [hunk 2]
```diff
@@ -235,8 +176,8 @@ class SavingsMakeTransferFragment : BaseFragment(), SavingsMakeTransferMvpView {
         transferDate = DateHelper.getSpecificFormat(DateHelper.FORMAT_dd_MMMM_yyyy,
                 getTodayFormatted())
         if (isLoanRepayment) {
-            amountField?.setText(outStandingBalance.toString())
-            amountFieldWrapper?.isFocusable = false
+            binding.amountField.setText(outStandingBalance.toString())
+            binding.amountFieldWrapper.isFocusable = false
         }
     }
```
Comment (gururani-abhishek) [inline app/src/main/java/org/mifos/mobile/ui/fragments/SavingsMakeTransferFragment.kt:183]: updateDetails() can be made private

File: app/src/main/java/org/mifos/mobile/ui/fragments/SavingsMakeTransferFragment.kt [hunk 3]
```diff
@@ -308,12 +249,12 @@ class SavingsMakeTransferFragment : BaseFragment(), SavingsMakeTransferMvpView {
...
 
     override fun showProgress() {
-        layoutMakeTransfer?.visibility = View.GONE
+        binding.llMakeTransfer?.visibility = View.GONE
```
Comment (gururani-abhishek) [inline app/src/main/java/org/mifos/mobile/ui/fragments/SavingsMakeTransferFragment.kt]: In "binding.llMakeTransfer?.visibility" please remove the unnecessary safe-call.

File: app/src/main/java/org/mifos/mobile/ui/fragments/SavingsMakeTransferFragment.kt [hunk 4]
```diff
@@ -325,61 +266,58 @@ class SavingsMakeTransferFragment : BaseFragment(), SavingsMakeTransferMvpView {
...
-        btnPayTo?.visibility = View.GONE
-        btnPayFrom?.visibility = View.VISIBLE
-        payFromFieldWrapper?.visibility = View.VISIBLE
-        payToFieldWrapper?.isEnabled = false
+        binding.processOne.setCurrentCompleted()
+        binding.processTwo.setCurrentActive()
+        binding.btnPayTo.visibility = View.GONE
+        binding.btnPayFrom.visibility = View.VISIBLE
+        binding.payFromFieldWrapper.visibility = View.VISIBLE
+        binding.payToFieldWrapper?.isEnabled = false
```
Comment (gururani-abhishek) [inline app/src/main/java/org/mifos/mobile/ui/fragments/SavingsMakeTransferFragment.kt]: safe-call can be removed here as well, "binding.payToFieldWrapper?.isEnabled"

File: app/src/main/java/org/mifos/mobile/ui/fragments/UpdatePasswordFragment.kt [hunk 5]
```diff
@@ -147,41 +144,44 @@ import javax.inject.Inject
...
                     getString(R.string.confirm_password),
                     resources.getInteger(R.integer.password_minimum_length))
             return false
         }
-        tilConfirmNewPassword?.isErrorEnabled = false
+        binding.tilConfirmNewPassword.isErrorEnabled = false
         return true
     }
 
+    override fun onDestroy() {
```
Comment (gururani-abhishek) [inline app/src/main/java/org/mifos/mobile/ui/fragments/UpdatePasswordFragment.kt]: This onDestroy() callback is redundant and can be removed.

**Review by gururani-abhishek — CHANGES_REQUESTED** [review]
Please have a look at these requested changes, rest it looks good to me. Good Job buddy!!

**Comment by PratyushSingh07** [comment]
made the changes that were asked in the review , do review this from your side as well @jawidMuhammadi

**Review by jawidMuhammadi — APPROVED** [review]
LGTM! :)

**Pull request merged**